.github/workflows/pypi-publish.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Upload Python Package
+
+on:
+  release:
+    types: [published]
+
+permissions:
+  contents: read
+
+jobs:
+  deploy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install build
+    - name: Build package
+      run: python -m build
+    - name: Publish package
+      uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
```

Comment:
@mateuslatrova  how does it know the correct package name? is it associated with the token? Have you tried uploading it to some dummy test package? PYPI has a test "version" for this sort of thing

Reply:
@alan-barzilay the package name is gotten from the "`name`" parameter in the "`setup`" function call inside "`setup.py`" file, which in this case is _pipreqs_.

yes, i have tested it in my PyPi account. I ran this pipeline for my fork of the pipreqs repo, and it worked.

if you want to, we can schedule a call for me to show you how it worked, and also what you will need to do for this to work in production